crates/ruff_cli/src/commands/run.rs

```diff
@@ -80,8 +81,10 @@ pub(crate) fn run(
     // Load the caches.
     let caches = bool::from(cache).then(|| {
         package_roots
-            .par_iter()
-            .map(|(package_root, _)| {
+            .values()
+            .flatten()
+            .dedup()
```

Comment:
Using `.dedup` here is subtly wrong because these may not be ordered, but it's fine, it's just an optimization. Will fix in a follow-up.